File: sdk/typescript/src/namespaces/auditing.ts
/**
 * Auditing Namespace API
 *
 * Provides deep audit, capability probing, and red team analysis capabilities.
 */

export interface CapabilityProbeRequest {
  agent: string;
  task: string;
  config?: Record<string, unknown>;
}

export interface CapabilityProbeResult {
  agent: string;
  task: string;
  score?: number;
  capabilities?: Record<string, unknown>;
  weaknesses?: string[];
  recommendations?: string[];
  metadata?: Record<string, unknown>;
}

export interface DeepAuditRequest {
  task: string;
  agents?: string[];
  depth?: 'shallow' | 'standard' | 'deep';
  config?: Record<string, unknown>;
}

export interface DeepAuditResult {
  audit_id: string;
  task: string;
  findings: Array<{
    category: string;
    severity: 'low' | 'medium' | 'high' | 'critical';
    description: string;
    evidence?: string;
  }>;
  summary: string;
  metadata?: Record<string, unknown>;
}

export interface RedTeamRequest {
  attack_types?: string[];
  intensity?: 'low' | 'medium' | 'high';
  config?: Record<string, unknown>;
}

export interface RedTeamResult {
  debate_id: string;
  attacks_run: number;
  vulnerabilities_found: number;
  findings: Array<{
    attack_type: string;
    success: boolean;
    description: string;
    severity?: string;
  }>;
  summary: string;
}

export interface AttackType {
  id: string;
  name: string;
  description: string;
  category: string;
}

interface AuditingClientInterface {
  request<T = unknown>(
    method: string,
    path: string,
    options?: { body?: unknown }
  ): Promise<T>;
}

export class AuditingAPI {
  constructor(private client: AuditingClientInterface) {}

  /**
   * Run capability probes on an agent.
   */
  async probeCapability(body: CapabilityProbeRequest): Promise<CapabilityProbeResult> {
    return this.client.request('POST', '/api/v1/debates/capability-probe', { body });
  }

  /**
   * Run a deep audit on a task.
   */
  async deepAudit(body: DeepAuditRequest): Promise<DeepAuditResult> {
    return this.client.request('POST', '/api/v1/debates/deep-audit', { body });
  }

  /**
   * Run red team analysis on a debate.
   */
  async redTeam(debateId: string, body?: RedTeamRequest): Promise<RedTeamResult> {
    return this.client.request('POST', `/api/v1/debates/${debateId}/red-team`, { body });
  }

  /**
   * Get available red team attack types.
   */
  async getAttackTypes(): Promise<{ attack_types: AttackType[] }> {
    return this.client.request('GET', '/api/v1/redteam/attack-types');
  }
}

File: sdk/python/aragora_sdk/namespaces/auditing.py
"""
Auditing Namespace API

Provides deep audit, capability probing, and red team analysis capabilities:
- Probe agent capabilities for specific tasks
- Run deep audits on tasks
- Perform red team analysis on debates
- Get available attack types
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any, Literal

if TYPE_CHECKING:
    from ..client import AragoraAsyncClient, AragoraClient

AuditDepth = Literal["shallow", "standard", "deep"]
AttackIntensity = Literal["low", "medium", "high"]
Severity = Literal["low", "medium", "high", "critical"]

class AuditingAPI:
    """
    Synchronous Auditing API.

    Provides deep audit, capability probing, and red team analysis.

    Example:
        >>> client = AragoraClient(base_url="https://api.aragora.ai")
        >>> result = client.auditing.probe_capability(
        ...     agent="claude",
        ...     task="Code review for security vulnerabilities"
        ... )
        >>> print(result["score"], result["capabilities"])
    """

    def __init__(self, client: AragoraClient):
        self._client = client

    def probe_capability(
        self,
        agent: str,
        task: str,
        config: dict[str, Any] | None = None,
    ) -> dict[str, Any]:
        """
        Run capability probes on an agent.

        Args:
            agent: The agent identifier to probe
            task: The task to test the agent's capabilities on
            config: Optional configuration for the probe

        Returns:
            Capability probe result including score, capabilities,
            weaknesses, and recommendations
        """
        data: dict[str, Any] = {"agent": agent, "task": task}
        if config:
            data["config"] = config

        return self._client.request("POST", "/api/v1/debates/capability-probe", json=data)

    def deep_audit(
        self,
        task: str,
        agents: list[str] | None = None,
        depth: AuditDepth = "standard",
        config: dict[str, Any] | None = None,
    ) -> dict[str, Any]:
        """
        Run a deep audit on a task.

        Args:
            task: The task to audit
            agents: Optional list of agents to include in the audit
            depth: Audit depth (shallow, standard, deep)
            config: Optional configuration for the audit

        Returns:
            Deep audit result including audit_id, findings, and summary
        """
        data: dict[str, Any] = {"task": task, "depth": depth}
        if agents:
            data["agents"] = agents
        if config:
            data["config"] = config

        return self._client.request("POST", "/api/v1/debates/deep-audit", json=data)

    def get_attack_types(self) -> dict[str, Any]:
        """
        Get available red team attack types.

        Returns:
            Dictionary containing list of available attack types
            with their id, name, description, and category
        """
        return self._client.request("GET", "/api/v1/redteam/attack-types")

class AsyncAuditingAPI:
    """
    Asynchronous Auditing API.

    Provides deep audit, capability probing, and red team analysis.

    Example:
        >>> async with AragoraAsyncClient(base_url="https://api.aragora.ai") as client:
        ...     result = await client.auditing.probe_capability(
        ...         agent="claude",
        ...         task="Code review for security vulnerabilities"
        ...     )
        ...     print(result["score"])
    """

    def __init__(self, client: AragoraAsyncClient):
        self._client = client

    async def probe_capability(
        self,
        agent: str,
        task: str,
        config: dict[str, Any] | None = None,
    ) -> dict[str, Any]:
        """
        Run capability probes on an agent.

        Args:
            agent: The agent identifier to probe
            task: The task to test the agent's capabilities on
            config: Optional configuration for the probe

        Returns:
            Capability probe result including score, capabilities,
            weaknesses, and recommendations
        """
        data: dict[str, Any] = {"agent": agent, "task": task}
        if config:
            data["config"] = config

        return await self._client.request("POST", "/api/v1/debates/capability-probe", json=data)

    async def deep_audit(
        self,
        task: str,
        agents: list[str] | None = None,
        depth: AuditDepth = "standard",
        config: dict[str, Any] | None = None,
    ) -> dict[str, Any]:
        """
        Run a deep audit on a task.

        Args:
            task: The task to audit
            agents: Optional list of agents to include in the audit
            depth: Audit depth (shallow, standard, deep)
            config: Optional configuration for the audit

        Returns:
            Deep audit result including audit_id, findings, and summary
        """
        data: dict[str, Any] = {"task": task, "depth": depth}
        if agents:
            data["agents"] = agents
        if config:
            data["config"] = config

        return await self._client.request("POST", "/api/v1/debates/deep-audit", json=data)

    async def get_attack_types(self) -> dict[str, Any]:
        """
        Get available red team attack types.

        Returns:
            Dictionary containing list of available attack types
            with their id, name, description, and category
        """
        return await self._client.request("GET", "/api/v1/redteam/attack-types")
